### DISCUSSION TITLE
ordering

### DISCUSSION THREAD

**Discussion by crasty01** [supabase/supabase#6513] [Questions] [answered]
Hi, I would like to ask, if there is any way to do something like this:

```SQL
select * from users
where username ILIKE '%del%'
order by username not ilike 'del%', length(username), username;
```

using supabase-js library.

Thanks in advance for any help.

**Comment by steve-chavez** [comment, marked as answer]
Not directly(only columns and computed columns are admitted as order terms in [supabase-js order()](https://supabase.com/docs/reference/dart/order)), but you can create a function with your exact query and then call it through [rpc](https://supabase.com/docs/reference/javascript/rpc).

**Reply by crasty01** [reply]
thanks 👍
